### ISSUE TITLE
Issues after TF 2.0 Beta Migration

### ISSUE THREAD

**Issue by lalo-remo93** [Raj-08/tensorflow-object-contour-detection#11]
Hi Raj,

First of all, thank you very much for sharing your code, it has been really helpful!

After performing as few modifications as I could, I was able to run it using TF 2.0. Unfortunately, I am not getting the expected results. 

In order to debug it, I am trying to use a small training set consisting of the two .jpg images you kindly provided. My debugging methodology is as follows:

1. Train the model during 101 epochs, using images 000129.png and 000999.png (which you also provided) as labels.
2. For monitoring purposes, save checkpoints and training predictions every 10 epochs.
3. After training is completed, make `is_training = False` in model_contour.py.
4. Evaluate model with eval.py using the last checkpoint ` 'U:\\logs\\model.ckpt-100'` 

After the model evaluation, I am obtaining float32 evaluation predictions with shape `(1, 480, 480, 1)` such as:

     [[[[0.5000359 ],
         [0.5002269 ],
         [0.49975395],
         ...,
         [0.5002495 ],
         [0.4999373 ],
         [0.5002173 ]],

        [[0.50023586],
         [0.5005435 ],
         [0.49990207],
         ...,
         [0.49997765],
         [0.5000773 ],
         [0.500376  ]],

        [[0.4999181 ],
         [0.5005329 ],
         [0.50022507],
         ...,
         [0.50042677],
         [0.4997322 ],
         [0.50028807]],

        ...,

        [[0.4996726 ],
         [0.49994084],
         [0.49953213],
         ...,
         [0.50001043],
         [0.50006956],
         [0.5002987 ]],

        [[0.5000668 ],
         [0.5000908 ],
         [0.49993053],
         ...,
         [0.50024533],
         [0.500082  ],
         [0.50001514]],

        [[0.50008976],
         [0.4999673 ],
         [0.4999259 ],
         ...,
         [0.50016195],
         [0.4999823 ],
         [0.5001097 ]]]]

Which results in a black image.

On the other hand, the prediction generated during training at epoch 100 looks like this:

![image](https://user-images.githubusercontent.com/52827748/61136994-7e394480-a4bc-11e9-9fcb-65c83b37ac0f.png)

As far as I understand, it should look very similar to the label, correct?

I was wondering if you have any idea on what could be wrong?

I can share with you the log of all the modifications I had to make to your code so that it can run in TF 2.0. Also, I am running TF in Windows 10 using a CPU.

Do you have any recommendation for debugging in this case?

Thank you in advance!

**Comment by lalo-remo93** [comment]
Hi Raj,

I was able to obtain a much better result by changing the learning rate to 0.0001:

![image](https://user-images.githubusercontent.com/52827748/61226033-a2de2800-a719-11e9-9d79-6770e98e17b8.png)

At this point, the total_loss was 5977.9404.

Do you think it is just a matter of increasing the number of training epochs to obtain an optimal result?

I also have a couple of questions about your implementation of the architecture presented in the paper:

1. Why are all `relu=False` in model_contour.py when the paper suggests a ReLU activation function to be used at least in all the decoder layers except the last one?

2. Shall this 2 lines remain commented at the end of model_contour.py for the code to work as expected:

`# bn_deconv0=batch_normalization(deconv0,is_training=is_training, name='bn_deconv0')`
`# pred = sigmoid(bn_deconv0,name='pred')`

?
